lib/api/contracts/model_contract.rb

```diff
@@ -51,6 +51,27 @@ def self.attribute(*attributes, &block)
       validate :readonly_attributes_unchanged
       validate :run_attribute_validations
 
+      # Performs validation of both, the contract and the underlying model and returns
+      # the merged validation errors for both of them.
+      def validate_contract_and_model
```

Comment:
So basically your idea is a different one than mine.
In my world contract and model are quite distinct. I just wanted to get this already existing merge-method to a more apropriate place and avoid that one `errors` object bleeds into the other one.

Your approach is different, you say "yes it is sad, that we have to validate both the contract and the model". But then you go on to say "but the caller does not need to know, we make him believe he only validates the contract".
This is vastly different from what I came up with, but I tell you: ***I LIKE IT!***

My course of action would now be to:

* call this baby `validate` and make it behave like the `super.validate` w.r.t return values and stuff
* merge the errors of the model and the contract into the *contract errors*
* let the caller believe it is only validating the contract alone

Benefits being that as soon as the day arrives when we validate via contracts only, we just need to remove this sweet little method override.

Do you agree? Did I understand you correctly?